### DISCUSSION TITLE
'Typed property must not be accessed before initialization' when using Form objects and rules() method

### DISCUSSION THREAD

**Discussion by richardwyke** [livewire/livewire#6620] [1. Help]
I have a relatively common scenario of validation for uniqueness on edit, This is working fine with Form objects for the create method, however on edit i don't see how to create this for an edit scenario where you need access to the ID of the model when running the validation rule

My livewire component looks like this:
```php
class ParticipantEdit extends Component
{
    public EditParticipantForm $form;

    public Participant $participant;

    public function mount(Participant $participant)
    {
        $this->participant  = $participant;

        $this->form->setParticipant($participant);
...
```
And the Form object looks like this:
```php
class EditParticipantForm extends Form
{
    public ?Participant $participant;

    public string $name;

    public function setParticipant(Participant $participant)
    {
        $this->participant = $participant;
    }

    public function rules()
    {
        return [
            'name'        => 'required|max:128',
            'reference'         => [
                'required',
                Rule::unique('participants', 'reference')
                    ->where(function (Builder $query) {
                        $query->where('organisation_id', active_org_id());
                    })
                    ->ignore($this->participant->id)
            ],
        ];
    }
}
```

However when this component is loaded I get this exception because the rules array is run once when the class is loaded and that's before ->setParticipant() is called in the parent component
`Typed property Rw\Packages\Core\Participants\Livewire\Employer\Forms\EditParticipantForm::$participant must not be accessed before initialization`

**Comment by basakabhi007** [comment]
did you find any solution for this? I am also facing this issue.

**Comment by pena-daniel** [comment]
Check your folder name i have the same problem the issue was that i were have a mistake in my folder name. in local your code editor is not case sensitive but on a machine the system is case sensitive.
